File: dotnet/PoeLogsParser/Parsers/PlayerJoinedAreaParser.cs
using System.Collections.Generic;
using System.Linq;
using System.Text.RegularExpressions;
using PoeLogsParser.Enums;
using PoeLogsParser.Models;
using PoeLogsParser.Models.Abstractions;

namespace PoeLogsParser.Parsers
{
    public class PlayerJoinedAreaParser : Parser
    {
        private readonly List<Regex> _regsClean = new List<Regex>()
        {
            new Regex(": ", RegexOptions.IgnoreCase),
            new Regex(" has joined the area.", RegexOptions.IgnoreCase)
        };

        private readonly Regex _regMatch = new Regex(".+ has joined the area.", RegexOptions.IgnoreCase);


        public override bool CanParse(string line)
        {
            return _regMatch.IsMatch(line);
        }

        public override ILogEntry Parse(string line, LogEntry entry)
        {
            return new PlayerJoinedAreaLogEntry(entry)
            {
                Types = new[] {LogEntryType.System, LogEntryType.Party, LogEntryType.JoinArea},
                Player = line
            };
        }

        public override string Clean(string line)
        {
            return _regsClean.Aggregate(line, (current, reg) => reg.Replace(current, ""));
        }
    }
}

File: ts/src/parsers/PlayerJoinedAreaParser.ts
import { LogEntryType } from "../enums/LogEntryType";
import { LogEntry } from "../models/entries/LogEntry";
import { PlayerJoinedAreaLogEntry } from "../models/entries/PlayerJoinedAreaLogEntry";
import { IParser } from "./abstractions/IParser";

export class PlayerJoinedAreaParser implements IParser {
	private _regMatch: RegExp = /.+ has joined the area./gi
	private _regsClean: RegExp[] = [
		/: /gi,
		/ has joined the area./gi
	];

	constructor() {
	}

	canParse(line: string): boolean {
		return this._regMatch.test(line);
	}

	parse(line: string, entry: LogEntry): PlayerJoinedAreaLogEntry {
		let playerJoinedAreaLogEntry = new PlayerJoinedAreaLogEntry(entry);

		playerJoinedAreaLogEntry.types = [LogEntryType.System, LogEntryType.Party, LogEntryType.JoinArea];

		playerJoinedAreaLogEntry.player = line;
		return playerJoinedAreaLogEntry;
	}

	clean(line: string): String {
		this._regsClean.forEach((reg) => line = line.replace(reg, ""));
		return line;
	}

	execute(entry: LogEntry, line: string): LogEntry {
		const cleanedline = this.clean(line);
		return this.parse(cleanedline as string, entry);
	}
}